[PR Port] Port #53882 - Fixed Bug LVM has no Filesystem Type , no Filesystem is not allowed

piterpunk opened the pull request:
### What does this PR do?
Port #53882 to master:

- Allow the creation of a partition without filesystem (for LVM, ASM, etc)
- Added unit tests to check if the mkpart allows the empty fs_type case.

Removed mkpartfs:

- Removed mkpartfs which was exactly the same function of mkpart, with an slightly different code which calls the same parted command with the same parameters. It's not used by other execution or state modules.

### What issues does this PR fix or reference?
Fixes: #54456

### Previous Behavior
An error when calling mkpart without fs_type specified

### New Behavior
When mkpart is called without a fs_type, the partition is created. 

### Merge requirements satisfied?
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

tyl0re committed:
Fixed Bug LVM has no Parttion Type.

The script later it is checked

 if fs_type:
        cmd = ('parted', '-m', '-s', '--', device, 'mkpart', part_type, fs_type, start, end)
    else:
        cmd = ('parted', '-m', '-s', '--', device, 'mkpart', part_type, start, end)

But never reached. The Check was in earlier Versions with parted.py

443     if fs_type and fs_type not in set(['ext2', 'fat32', 'fat16', 'linux-swap', 'reiserfs',
444                           'hfs', 'hfs+', 'hfsx', 'NTFS', 'ufs', 'xfs', 'zfs']):

So the check on not defined fs_type is missing

Commit:
Added mkpart unit tests and removed mkpartfs

- Removed mkpartfs as it is the same of mkpart. No code inside Salt
  uses mkpartfs.
- Moved the last change "if fs_type and not _is_fstype(fs_type)" to
  mkpart.
- Added tests to check if mkpart allows the creation of a partition
  with no fstype specified.

Commit:
Added changelog entry

Ch3LL commented on salt/modules/parted_partition.py:
```diff
@@ -542,39 +545,6 @@ def mkpart(device, part_type, fs_type=None, start=None, end=None):
     return out
 
 
-def mkpartfs(device, part_type, fs_type, start, end):
```
We cannot remove this function, because a user could be using this even if its not being called from another state/execution module. Could you possibly have this function call the other to ensure backwards compatibility?

piterpunk replied:
Of course. I will do that tonight.

piterpunk replied:
Re-added the mkpartfs which now is just a call to mkpart. Also added a test to check if all arguments from mkpartfs are reaching mkpart.

Commit:
Re-added mkpartfs to not break compatibility

- Re-added mkpartfs to not break compatibility, now it only calls
  mkpart.
- Added a new unit test to make sure that mkpartfs is still there
  and all arguments are passed to mkpart

Ch3LL commented on salt/modules/parted_partition.py:
```diff
@@ -542,36 +545,19 @@ def mkpart(device, part_type, fs_type=None, start=None, end=None):
+        salt '*' partition.mkpartfs /dev/sda primary fs_type=fat32 start=0 end=639
+        salt '*' partition.mkpartfs /dev/sda primary start=0 end=639
     """
-    _validate_device(device)
```
thanks for updating this. looks like this is the difference between the two. Is there a reason we dont want to add this to `mkpart`?

piterpunk replied:
No reason at all. I just re-added the _validate_device and updated the tests accordingly.

Ch3LL reviewed (changes requested):
just one more question here, everything else looks great thanks

Commit:
Added device validation before executing mkpart
